chains/lisk/client.go

```diff
@@ -0,0 +1,113 @@
+package lisk
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/sisu-network/deyes/chains/lisk/types"
+	"github.com/sisu-network/deyes/config"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"reflect"
+	"strconv"
+)
+
+type APIErr struct {
+	message string
+}
+
+func NewApiErr(message string) error {
+	return &APIErr{message: message}
+}
+
+func (e *APIErr) Error() string {
+	return fmt.Sprintf("cannot fetch data %s", e.message)
+}
+
+// LiskClient  A wrapper around lisk.client so that we can mock in watcher tests.
+type LiskClient interface {
+	BlockNumber() (uint64, error)
+	BlockByHeight(height uint64) (*types.Block, error)
+	TransactionByBlock(block string) ([]types.Transaction, error)
+}
+
+type defaultLiskClient struct {
+	chain   string
+	baseApi string
+}
+
+func NewLiskClient(cfg config.Chain) LiskClient {
+	c := &defaultLiskClient{
+		chain:   cfg.Chain,
+		baseApi: cfg.BaseApi,
+	}
+	return c
+}
+
+func (c *defaultLiskClient) execute(endpoint string, params map[string]string) ([]map[string]interface{}, error) {
+	keys := reflect.ValueOf(params).MapKeys()
+	req, err := http.NewRequest("GET", c.baseApi+endpoint, nil)
+	if err != nil {
+		log.Print(err)
+	}
+	q := req.URL.Query()
+	for _, key := range keys {
+		q.Add(key.Interface().(string), params[key.Interface().(string)])
+	}
+
+	req.URL.RawQuery = q.Encode()
+	response, err := http.Get(req.URL.String())
+	if response == nil {
+		return nil, NewApiErr(endpoint)
+	}
+	responseData, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var responseObject types.ResponseWrapper
+	json.Unmarshal(responseData, &responseObject)
+	return responseObject.Data, err
+}
+
+func (c *defaultLiskClient) BlockNumber() (uint64, error) {
+	params := map[string]string{
+		"limit": "1",
+		"sort":  "height:desc",
+	}
+	blocks, err := c.execute("/blocks", params)
+	latestBlock := blocks[0]
+	latestBlockJson, _ := json.Marshal(latestBlock)
+	var block types.Block
+	if err := json.Unmarshal(latestBlockJson, &block); err != nil {
```

Comment:
What's going on here? Why do you marshal & unmarshal?

Reply:
I need to convert from  type interface{} to type.Block. the first I convert Interface{} to JSON then  JSON to type.Block
Can you tell me what is best way to convert interface{} to type.Block